client/my-sites/people/team-invite/invite-form.tsx

```diff
@@ -63,45 +64,60 @@ function InviteForm( props: Props ) {
 	const [ showContractorCb, setShowContractorCb ] = useState( false );
 	const [ readyForSubmit, setReadyForSubmit ] = useState( false );
 
-	useEffect( extendTokenFormControls, [ tokenValues ] );
-	useEffect( toggleShowContractorCb, [ role ] );
-	useEffect( checkSubmitReadiness, [ tokenErrors, validationProgress ] );
-	useEffect( reactOnInvitationSuccess, [ invitingSuccess ] );
+	useEffect( extendTokenFormControls, [ tokenControlNum, tokenValues ] );
+	useEffect( toggleShowContractorCb, [ isSiteForTeams, role ] );
+	useEffect( checkSubmitReadiness, [
+		readyForSubmit,
+		tokenErrors,
+		tokenValues,
+		validationProgress,
+	] );
+	const resetFormValues = useCallback( () => {
+		setRole( defaultUserRole );
+		setTokenValues( [ '' ] );
+		setContractor( false );
+		setMessage( '' );
+	}, [ defaultUserRole ] );
+
+	useEffect( reactOnInvitationSuccess, [ invitingSuccess, onInviteSuccess, resetFormValues ] );
 	useEffect( () => {
 		prevInvitingProgress.current = invitingProgress;
 	}, [ invitingProgress ] );
 	useValidationNotifications();
 	useInvitingNotifications( tokenValues );
 
-	function onFormSubmit( e: FormEvent ) {
-		e.preventDefault();
-		if ( ! readyForSubmit || invitingProgress ) {
-			return;
-		}
+	const onFormSubmit = useCallback(
```

Comment:
Is there any specific reason to add a `useCallback`  here?

I noticed that `onTokenChange` or `reactOnInvitationSuccess` are not wrapped with `useCallback`.

If it's not strictly necessary to fix this issue I would remove the `useCallback` for `onFormSubmit`.

Reply:
Thanks for the review @sejas - I don't think it is strictly necessary. If I remember correctly, we added it to ensure that it renders only when dependencies update but it works without it as well. I removed it as per your suggestion.

As for the error being shown after the button is clicked, it is somewhat unavoidable because we start validation 3s after the user starts typing so it finishes with the delay. I decreased the time in `useDebouncedCallback` to a bit shorter to make this less visible but it is still going to be there. Otherwise, if we remove it completely, we will be sending API requests on every keystroke.